pkg/knuu/instance.go

```diff
@@ -16,35 +16,7 @@ import (
 	rbacv1 "k8s.io/api/rbac/v1"
 )
 
-// Instance represents a instance
-type Instance struct {
-	name                     string
-	imageName                string
-	k8sName                  string
-	state                    InstanceState
-	instanceType             InstanceType
-	kubernetesService        *v1.Service
-	builderFactory           *container.BuilderFactory
-	kubernetesStatefulSet    *appv1.StatefulSet
-	portsTCP                 []int
-	portsUDP                 []int
-	command                  []string
-	args                     []string
-	env                      map[string]string
-	volumes                  []*k8s.Volume
-	memoryRequest            string
-	memoryLimit              string
-	cpuRequest               string
-	policyRules              []rbacv1.PolicyRule
-	livenessProbe            *v1.Probe
-	readinessProbe           *v1.Probe
-	startupProbe             *v1.Probe
-	files                    []*k8s.File
-	isSidecar                bool
-	parentInstance           *Instance
-	sidecars                 []*Instance
-	ingress                  *Ingress
-	externalDns              []string
+type ObsyConfig struct {
```

Comment:
[proposal][feature-request]: please make a dedicated issue that tackles refactoring of obsyconfig to more composable structs per type of stack we use for metrics/traces (otlp, prom, jaeger, pyro, etc.)

Reply:
https://github.com/celestiaorg/knuu/issues/178